packages/client/src/__tests__/dataTriggers.test.ts

```diff
@@ -674,4 +675,177 @@ describe("Cloud functions -> Data triggers ->", () => {
+            .set(slotNextMonthIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateThisMonth}-10`))
+            .set(slotThisMonthOffIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateNextMonth}-10`))
+            .set(slotNextMonthOffIce),
+          adminDb.doc(getCustomerDocPath(organization, saul.id)).set(saul),
+        ]);
+
+        await Promise.all([
+          await waitFor(async () => {
+            const snap = await adminDb
+              .doc(getCustomerDocPath(organization, saul.id))
+              .get();
+            expect(snap.data()).toEqual(saul);
+          }),
+        ]);
+
+        const bookedSlotThisMonthIce = {
+          date: slotThisMonthIce.date,
+          interval: Object.keys(slotThisMonthIce.intervals)[0],
+        };
+        const bookedSlotNextMonthIce = {
+          date: slotNextMonthIce.date,
+          interval: Object.keys(slotNextMonthIce.intervals)[0],
+        };
+
+        const bookedSlotThisMonthOffIce = {
+          date: slotThisMonthOffIce.date,
+          interval: Object.keys(slotThisMonthOffIce.intervals)[0],
+        };
+        const bookedSlotNextMonthOffIce = {
+          date: slotNextMonthOffIce.date,
+          interval: Object.keys(slotNextMonthOffIce.intervals)[0],
+        };
+
+        // book slots
+        await Promise.all([
+          await adminDb
+            .doc(
+              getBookedSlotDocPath(
+                organization,
+                saul.secretKey,
+                `${bookedSlotThisMonthIce.date}-9`
+              )
+            )
+            .set(bookedSlotThisMonthIce),
+          await adminDb
+            .doc(
+              getBookedSlotDocPath(
+                organization,
+                saul.secretKey,
+                `${bookedSlotNextMonthIce.date}-9`
+              )
+            )
+            .set(bookedSlotNextMonthIce),
+          await adminDb
+            .doc(
+              getBookedSlotDocPath(
+                organization,
+                saul.secretKey,
+                `${bookedSlotThisMonthOffIce.date}-10`
+              )
+            )
+            .set(bookedSlotThisMonthOffIce),
+          await adminDb
+            .doc(
+              getBookedSlotDocPath(
+                organization,
+                saul.secretKey,
+                `${bookedSlotNextMonthOffIce.date}-10`
+              )
+            )
+            .set(bookedSlotNextMonthOffIce),
+        ]);
+
+        // The booked slot should be created in customer's bookings
+        await waitFor(async () => {
+          const snap = await adminDb
+            .doc(
+              getBookedSlotDocPath(
+                organization,
+                saul.secretKey,
+                `${bookedSlotThisMonthIce.date}-9`
+              )
+            )
+            .get();
+
+          expect(snap.data()).toEqual(bookedSlotThisMonthIce);
+        });
```

Comment:
Same as when adding customer: If we explicitly store the doc, the doc is there when the `.set` promise resolves, no need for additional check.